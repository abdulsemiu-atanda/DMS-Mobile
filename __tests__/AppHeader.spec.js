import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import {fromJS} from 'immutable'

import AppHeader from '../components/shared/AppHeader.react'

const props = {
  navigation: {
    state: {
      routes: []
    }
  }
}

const routeParamsProps = {
  navigation: {
    state: {
      routes: [
        {
          params: {
            documents: fromJS([{access: 'Private'}, {access: 'Private'}])
          }
        },
        {
          params: {
            documents: fromJS([{access: 'Private'}, {access: 'Private'}])
          }
        }
      ]
    }
  }
}

describe('AppHeader', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppHeader {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[0].children[0]).toEqual('DMS')
  })

  it('renders correctly when routes is not an empty array', () => {
    const tree = renderer.create(<AppHeader {...routeParamsProps} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[0].children[0]).toEqual('PRIVATE')
  })
})
