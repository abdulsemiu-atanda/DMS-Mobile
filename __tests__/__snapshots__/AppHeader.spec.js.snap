// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "height": 93.38000000000001,
      "justifyContent": "space-between",
      "width": 750,
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "600",
        "marginLeft": 20,
        "marginTop": 20,
      }
    }
  >
    DMS
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#0983ff",
              "fontSize": 25,
            },
            Object {
              "marginRight": 15,
              "marginTop": 15,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#0983ff",
              "fontSize": 30,
            },
            Object {
              "marginRight": 15,
              "marginTop": 15,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`AppHeader renders correctly when routes is not an empty array 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "height": 93.38000000000001,
      "justifyContent": "space-between",
      "width": 750,
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "600",
        "marginLeft": 20,
        "marginTop": 20,
      }
    }
  >
    PRIVATE
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#0983ff",
              "fontSize": 25,
            },
            Object {
              "marginRight": 15,
              "marginTop": 15,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#0983ff",
              "fontSize": 30,
            },
            Object {
              "marginRight": 15,
              "marginTop": 15,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
